src/reporter/index.js

```diff
@@ -177,6 +198,22 @@ export default class Reporter {
             await reportItem.pendingTestRunDonePromise;
         });
 
+        task.on('test-action-start', async ({ apiActionName, command, testRun }) => {
+            if (this.plugin.reportTestActionStart) {
+                const args = this._prepareReportTestActionEventArgs({ command, testRun });
+
+                await this.plugin.reportTestActionStart(apiActionName, args);
+            }
+        });
+
+        task.on('test-action-done', async ({ apiActionName, command, testRun, errors }) => {
+            if (this.plugin.reportTestActionDone) {
+                const args = this._prepareReportTestActionEventArgs({ command, testRun, errors });
+
+                await this.plugin.reportTestActionDone(apiActionName, args);
```

Comment:
Shouldn't we provide a default implementation for `reportTestActionStart` and `reportTestActionDone` in [src/reporter/plugin-host.js](https://github.com/DevExpress/testcafe/blob/master/src/reporter/plugin-host.js) or check for their existence?